### PR TITLE
docs: core-geth features, generated API method docs

### LINKED ISSUE #308: docs: add OpenRPC service discovery document markdown
Currently the best way (...erm, only way) to view the `rpc.discover` document is via the [OpenRPC Playground](playground.open-rpc.org/), eg. https://playground.open-rpc.org/?url=https://gist.githubusercontent.com/meowsbits/4da4c08765679dac1899543002d1f545/raw/spec.json

It'd be great to get these docs into generated markdown (via CI) so they can be included and served at https://etclabscore.github.io/core-geth.

### PR DESCRIPTION
**Pull request by meowsbits**
Working on adding some documentation about what makes core-geth core-geth.

Resolves #308

### CODE REVIEW

**Comment by meowsbits** [comment]
Nice to have:
- https://squidfunk.github.io/mkdocs-material/reference/content-tabs/
- https://github.com/ethereum/go-ethereum/pull/21338 (see `MethodInvocation` section, is nice)

File: ethclient/ethclient_test.go [hunk 1]
```diff
@@ -419,6 +424,221 @@ func TestRPCDiscover(t *testing.T) {
...
+{{ define "contentDescTpl" -}}
+{{ $nameyDescription := trimNameSpecialChars .description }}
+{{ if eq .name $nameyDescription }}
+<code>{{ .description }}</code> {{ if .summary }}_{{ .summary }}_{{- end }}
+{{- else -}}
+{{ .name }} <code>{{ .description }}</code> {{ if .summary }}_{{ .summary }}_{{- end }}
+{{- end }}
+
+  + Required: {{ if .required }}✓ Yes{{ else }}No{{- end}}
+{{ if .deprecated }}  + Deprecated: :warning: Yes{{- end}}
```
Comment (meowsbits) [inline ethclient/ethclient_test.go]: Tabs
https://squidfunk.github.io/mkdocs-material/reference/content-tabs/

File: ethclient/ethclient_test.go [hunk 2]
```diff
@@ -419,6 +424,221 @@ func TestRPCDiscover(t *testing.T) {
...
+{{ if .Result -}}
+{{ $result := .Result | tomap }}
+{{- if ne $result.name "Null" }}
+{{ template "contentDescTpl" $result }}
+{{- else -}}
+_None_
+{{- end }}
+{{- end }}
+
+
```
Comment (meowsbits) [inline ethclient/ethclient_test.go]: Include client example Go, `curl`, `js console`

**Comment by meowsbits** [comment]
I'm gonna squash this and mark ready for review.

File: ethclient/ethclient_test.go (hunk 1 above)
Comment (meowsbits) [reply ethclient/ethclient_test.go]: Done.

File: ethclient/ethclient_test.go (hunk 2 above)
Comment (meowsbits) [reply ethclient/ethclient_test.go]: Done, but no `Go` example yet.

**Comment by meowsbits** [comment]
Aside from the generated API method docs (`./docs/JSON-RPC-API/modules/*.md`), here's the most significant content addition: https://github.com/etclabscore/core-geth/blob/docs/core-geth-features/docs/core/index.md.

`tree` for reference:

```
> tree docs
docs
├── audits
│   ├── 2017-04-25_Geth-audit_Truesec.pdf
│   ├── 2018-09-14_Clef-audit_NCC.pdf
│   ├── 2019-10-15_Discv5_audit_LeastAuthority.pdf
│   └── 2020-01-24_DiscV5_audit_Cure53.pdf
├── core
│   ├── alltools.md
│   ├── evmc.md
│   └── index.md
├── developers
│   ├── build-from-source.md
│   ├── create-new-release.md
│   ├── documentation.md
│   └── versioning.md
├── getting-started
│   ├── installation.md
│   └── run-cli.md
├── img
│   └── favicon.png
├── index.md
├── JSON-RPC-API
│   ├── index.md
│   ├── modules
│   │   ├── admin.md
│   │   ├── debug.md
│   │   ├── ethash.md
│   │   ├── eth.md
│   │   ├── miner.md
│   │   ├── net.md
│   │   ├── personal.md
│   │   ├── trace.md
│   │   ├── txpool.md
│   │   └── web3.md
│   └── openrpc.md
└── tutorials
    └── private-network.md

8 directories, 28 files
```

**Comment by meowsbits** [comment]
Just putting this out there... the generated docs in `docs/JSON-RPC-APIs/modules/` aren't deterministic. And this bothers me. It's going to be practically impossible to review diffs, for example. The nondeterminism is caused, I believe exclusively, by Go's non-guaranteed map ordering.

**Comment by ziogaschr** [comment]
Would be nice if they can be deterministic, though I can leave with that as this PR brings a lot of value in the docs.

**Comment by meowsbits** [comment]
https://github.com/etclabscore/core-geth/pull/306/commits/cc5afb07f9f0dc3ff86087a316f5c25953b60d9d begins the ordering solve. In fact, so far, it seems to be 100% effective. I don't think its a complete solution because the template also includes information (as `Raw JSON`) from the plain old `json.MarshalIndent` which shouldn't guarantee ordering, but at least for me, for now, that doesn't seem to be an issue. Note this commit's diff with https://github.com/etclabscore/core-geth/pull/306/commits/de739184a71d5082d777a6ea3ce2be61e621808a, showing that only the generated-at timestamps and expected lines are modified.

> Would be nice if they can be deterministic, though I can leave with that as this PR brings a lot of value in the docs.
> https://github.com/etclabscore/core-geth/pull/306#issuecomment-765666489

With these last two commits, I'm happy merging this and following up later if the `json.MarshalIndent` causes undesirable noise. For now, close enough.

**Comment by meowsbits** [comment]
@ziogaschr PTAL https://github.com/etclabscore/core-geth/pull/306#issuecomment-765905281. I'm OK with merging if you are.

**Comment by ziogaschr** [comment]
Looks fine for me at the moment. Nice work man. We can do a new PR when and if needed.

**Review by ziogaschr — APPROVED** [review]

**Pull request merged**